settings/OpeningPeriodForm/BigCalendarWrapper.js

```diff
@@ -122,7 +125,37 @@ class BigCalendarWrapper extends React.PureComponent {
       this.onCalendarChange(nextEvents);
     };
 
+  checkEventExistOrOverlap = (startTime, endTime) => {
+    const { events } = this.state;
+    const existingEventsArr = events.map(event => {
+      const { start, end } = event;
+      const startOfExistedEvent = moment(start);
+      const endOfExistedEvent = moment(end);
+      const startTimeMoment = moment(startTime);
+      const endTimeMoment = moment(endTime);
+      const dublicatedEvent = startTimeMoment.isSame(startOfExistedEvent) && endTimeMoment.isSame(endOfExistedEvent);
+      const startBeforeEvent = startTimeMoment.isBefore(startOfExistedEvent)
+        && endTimeMoment.isSameOrBefore(endOfExistedEvent)
+        && endTimeMoment.isAfter(startOfExistedEvent);
+      const startAfterEvent = startTimeMoment.isSameOrAfter(startOfExistedEvent)
+        && startTimeMoment.isBefore(endOfExistedEvent)
+        && (endTimeMoment.isSameOrAfter(endOfExistedEvent) || endTimeMoment.isSameOrBefore(endOfExistedEvent));
+      const fullOverlap = (startTimeMoment.isAfter(startOfExistedEvent) && endTimeMoment.isBefore(endOfExistedEvent))
+        || (startTimeMoment.isBefore(startOfExistedEvent) && endTimeMoment.isAfter(endOfExistedEvent));
+
+      if (dublicatedEvent || startBeforeEvent || startAfterEvent || fullOverlap) {
+        return false;
+      }
+      return true;
+    });
+
+    return existingEventsArr.every(event => event);
```

Comment:
By the way, I suggest replacing `map` call [here](https://github.com/folio-org/ui-calendar/pull/232/files#diff-5dad3b4739816352b6df03c31f641704R130) with `every` or `some`, so the code will be optimized when some of the condition is not met. 

Imagine, `events` has a size of ten elements and the first one is falsy. With `map`, all ten events will be processed before they reach `every` call here - with `every`/`some` there only one event will be processed.